deepchem/feat/tests/test_equivariant_graph_featurizer.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+from rdkit import Chem
+from deepchem.feat import EquivariantGraphFeaturizer
+from deepchem.feat.graph_data import GraphData
+import numpy as np
+
+
+class TestEquivariantGraphFeaturizer(unittest.TestCase):
+    """
+    Tests for EquivariantGraphFeaturizer.
+    """
+
+    def setUp(self):
+        """
+        Set up tests with a simple molecule.
+        """
+        smiles = 'CCO'
+        self.mol = Chem.MolFromSmiles(smiles)
+
+    def test_bonded_graph_featurization(self):
+        """
+        Test featurization with bonded edges.
+        """
+        featurizer = EquivariantGraphFeaturizer(fully_connected=False,
+                                                embeded=True)
+        graphs = featurizer.featurize([self.mol])
+
+        assert len(graphs) == 1
+        graph = graphs[0]
+
+        assert isinstance(graph, GraphData)
+
+        assert graph.node_features.shape[0] == self.mol.GetNumAtoms()
+        assert graph.positions.shape[0] == self.mol.GetNumAtoms()
+        assert graph.edge_index.shape[1] > 0
+
+    def test_fully_connected_graph_featurization(self):
+        """
+        Test featurization with fully connected edges.
+        """
+        featurizer = EquivariantGraphFeaturizer(fully_connected=True,
+                                                embeded=True)
+        graphs = featurizer.featurize([self.mol])
+        assert len(graphs) == 1
+
+        graph = graphs[0]
+        assert isinstance(graph, GraphData)
+
+        num_atoms = self.mol.GetNumAtoms()
+
+        expected_edges = num_atoms * (num_atoms - 1)  # Fully connected graph
+        assert graph.edge_index.shape[1] == expected_edges
+
```

Comment:
Can you add a unit test for testing distance between atoms? You can do that in a follow up PR